WalletWasabi/WabiSabi/WabiSabiCoordinator.cs

```diff
@@ -139,16 +141,46 @@ public void BanDoubleSpenders(object? sender, Transaction tx)
 		// Ban all outputs created by the received transaction because it has spent coins participating in coinjoin rounds.
 		foreach (var indexedOutput in tx.Outputs.AsIndexedOutputs())
 		{
-			Warden.Prison.DoubleSpent(new OutPoint(tx, indexedOutput.N), indexedOutput.TxOut.Value, disruptedRounds);
+			Warden.Prison.DoubleSpent(new OutPoint(tx, indexedOutput.N), indexedOutput.TxOut.Value, disruptedRounds.Select(x => x.RoundId));
 		}
 
-		// Abort disrupted rounds
-		foreach (var roundId in disruptedRounds)
+		// Abort disrupted rounds (only those that pay less than the attacking transaction)
+		using var cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(100));
+		var (succeed, spentCoins) = await GetSpendingCoinsAsync(inputOutPoints, cts.Token).ConfigureAwait(false);
+		var feeRate = succeed ? tx.GetFeeRate(spentCoins) : Constants.AbsurdlyHighFeeRate;
+		var roundsToAbort = disruptedRounds
+			.Where(round => round.MiningFeeRate < feeRate)
+			.Select(x => x.RoundId);
+
+		foreach (var roundId in roundsToAbort)
 		{
 			Arena.AbortRound(roundId);
 		}
 	}
 
+	private async Task<(bool, Coin[])> GetSpendingCoinsAsync(
+		IEnumerable<OutPoint> spendingOutPoints,
+		CancellationToken cancellationToken)
+	{
+		var batch = RpcClient.PrepareBatch();
+		var getTxOutRequests = spendingOutPoints
+			.Select(x => RpcClient.GetTxOutAsync(x.Hash, (int) x.N, true, cancellationToken))
+			.ToList();
+		await batch.SendBatchAsync(cancellationToken).ConfigureAwait(false);
+		var txOutResponses = await Task.WhenAll(getTxOutRequests).ConfigureAwait(false);
+		if (txOutResponses.Any(x => x is null))
+		{
+			return (false, Array.Empty<Coin>());
+		}
```

Comment:
How can a response here be `null`?
Is it safe to say then when `Any` of the responses are `null`, we simply disrupt the round?

Reply:
It is a good question. It would be good to add a comment to the code. It returns null if it is spent? https://developer.bitcoin.org/reference/rpc/gettxout.html

Reply:
No, it returns `null` is the output is not found.